test/backend/datastores.athena.spec.js

```diff
@@ -0,0 +1,327 @@
+import {assert} from 'chai';
+
+import {connect} from '../../backend/persistent/datastores/Athena';
+
+/* eslint-disable no-invalid-this */
+describe('Athena Unit Tests:', function () {
+    it('connect succeeds', function(done) {
+        const conn = {
+            region: 'us-east-1',
+            accessKey: 'XXXXXXXX',
+            secretKey: 'XXXXXAAAA',
+            database: 'PLOT.LY-TEST',
+            sqlStatement: 'SELECT * FROM TEST_TABLE LIMIT 100',
+            s3Outputlocation: 's3://aws-athena-query-results-11111111-us-east-1/',
+            queryTimeout: 10000
+        };
+        connect(conn).then(connection => {
+            assert.isDefined(connection, 'Connection is defined');
+            done();
+        }).catch(err => {
+            done(`Should have obtained a valid connection ${err}`);
+        });
+    });
+
+    it('connect fails missing region', function(done) {
+        const conn = {
+            accessKey: 'XXXXXXXX',
+            secretKey: 'XXXXXAAAA',
+            database: 'PLOT.LY-TEST',
+            sqlStatement: 'SELECT * FROM TEST_TABLE LIMIT 100',
+            s3Outputlocation: 's3://aws-athena-query-results-11111111-us-east-1/',
+            queryTimeout: 10000
+        };
+        try {
+            connect(conn).then(connection => {
+                assert.isUnDefined(connection, 'Connection should not have been defined');
+                done('Should have not have occured do to missing region');
+            }).catch(() => {
+                done();
+            });
+        } catch (error) {
+            done();
+        }
```

Comment:
If you want to skip a test, you just need to replace `it` by `xit`, e.g:
```js
xit('this spec will be skipped', function() {...});

it('this spec will be run', function() {...});

```

Reply:
I am not looking to skip these tests.  I am making sure that the validation is working correctly.  I am intentionally passing in bad data and I want to make sure that the connection function rejects the test.

Reply:
Unless I'm misreading it, this test won't fail if connection is `undefined`.